File: examples/tasks/hello.py
#!/usr/bin/env python3
"""
Simple Hello World task for Aido

This demonstrates the basics of an Aido task:
- Reading from inputs (if provided)
- Writing to outputs
- Logging to stdout
- Exiting with status code
"""

import sys
import os
from pathlib import Path
from datetime import datetime


def main():
    print("=" * 50)
    print("Aido Example Task: Hello World (Python)")
    print("=" * 50)
    print()

    # Show environment info
    print(f"Task started at: {datetime.now().isoformat()}")
    print(f"Python version: {sys.version}")
    print(f"Working directory: {os.getcwd()}")
    print()

    # Check for inputs
    inputs_dir = Path("/inputs")
    if inputs_dir.exists():
        print("📁 Inputs available:")
        for item in inputs_dir.rglob("*"):
            if item.is_file():
                print(f"  - {item.relative_to(inputs_dir)}")
    else:
        print("📁 No inputs directory found (running in local mode)")
    print()

    # Create output
    outputs_dir = Path("/outputs")
    if not outputs_dir.exists():
        outputs_dir = Path("./outputs")
        outputs_dir.mkdir(exist_ok=True)

    output_file = outputs_dir / "hello.txt"

    message = f"""Hello from Aido!

Generated at: {datetime.now().isoformat()}
Python version: {sys.version.split()[0]}

This is a simple example task that demonstrates:
✓ Execution in an isolated sandbox
✓ Reading from inputs (if provided)
✓ Writing to outputs
✓ Logging to stdout

Status: SUCCESS
"""

    output_file.write_text(message)
    print(f"✓ Output written to: {output_file}")
    print()

    # Success
    print("=" * 50)
    print("Task completed successfully!")
    print("=" * 50)
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: examples/tasks/hello.ts
#!/usr/bin/env node
/**
 * Simple Hello World task for Aido (TypeScript/Node.js)
 *
 * This demonstrates the basics of an Aido task:
 * - Reading from inputs (if provided)
 * - Writing to outputs
 * - Logging to stdout
 * - Exiting with status code
 */

import { writeFileSync, existsSync, readdirSync, statSync } from 'fs';
import { join } from 'path';

function main(): number {
  console.log('='.repeat(50));
  console.log('Aido Example Task: Hello World (TypeScript)');
  console.log('='.repeat(50));
  console.log();

  // Show environment info
  const now = new Date().toISOString();
  console.log(`Task started at: ${now}`);
  console.log(`Node version: ${process.version}`);
  console.log(`Working directory: ${process.cwd()}`);
  console.log();

  // Check for inputs
  const inputsDir = '/inputs';
  if (existsSync(inputsDir)) {
    console.log('📁 Inputs available:');
    const files = getAllFiles(inputsDir);
    files.forEach(file => {
      console.log(`  - ${file}`);
    });
  } else {
    console.log('📁 No inputs directory found (running in local mode)');
  }
  console.log();

  // Create output
  const outputsDir = existsSync('/outputs') ? '/outputs' : './outputs';
  const outputFile = join(outputsDir, 'hello.txt');

  const message = `Hello from Aido!

Generated at: ${now}
Node version: ${process.version}

This is a simple example task that demonstrates:
✓ Execution in an isolated sandbox
✓ Reading from inputs (if provided)
✓ Writing to outputs
✓ Logging to stdout

Status: SUCCESS
`;

  writeFileSync(outputFile, message, 'utf-8');
  console.log(`✓ Output written to: ${outputFile}`);
  console.log();

  // Success
  console.log('='.repeat(50));
  console.log('Task completed successfully!');
  console.log('='.repeat(50));
  return 0;
}

function getAllFiles(dir: string, base = ''): string[] {
  const files: string[] = [];
  const items = readdirSync(dir);

  for (const item of items) {
    const fullPath = join(dir, item);
    const relativePath = join(base, item);

    if (statSync(fullPath).isDirectory()) {
      files.push(...getAllFiles(fullPath, relativePath));
    } else {
      files.push(relativePath);
    }
  }

  return files;
}

process.exit(main());
